modules/lmdbbackend/lmdbbackend.cc

```diff
@@ -1680,6 +1715,44 @@ bool LMDBBackend::updateEmptyNonTerminals(uint32_t domain_id, set<DNSName>& inse
   return false;
 }
 
+bool LMDBBackend::superMasterAdd(const string& ip, const string& nameserver, const string& account)
+{
+  auto txn = d_tautoprimaries->getRWTransaction();
+  /* check if it exists first */
+  DNSName ns(nameserver);
+  for (auto range = txn.equal_range<0>(ip); range.first != range.second; ++range.first)
+    if (range.first->nameserver == ns)
+      range.first.del();
```

Comment:
so .. why is it not working? I copypasted it from elsewhere in the lmdb code.

Reply:
resolved this by iterating all autoprimaries and matching in code. not very pretty, but then again, there isn't so many autoprimaries that this would concern. I did not figure out a way to get all matching entries with equal_range, it seems to only return one.